framework/src/st_things/things_stack/framework/things_data_manager.c

```diff
@@ -345,9 +345,33 @@ static struct st_resource_type_s *create_resource_type()
 
 static void delete_resource_type(st_resource_type_s *type)
 {
+	if (type == NULL) {
+		return;
+	}
+
+	int count = type->prop_cnt;
+	for (int i = 0; i < count; ++i) {
+		things_free(type->prop[i]);
+	}
 	things_free(type);
 }
 
+static void delete_resource_type_map()
+{
+	if (g_resource_type_hmap != NULL) {
+		unsigned long *keyset = hashmap_get_keyset(g_resource_type_hmap);
+		long key_cnt = hashmap_count(g_resource_type_hmap);
+		for (long type_iter = 0; type_iter < key_cnt; ++type_iter) {
```

Comment:
Some compilers does not understand define a datatype in the for statement.

Reply:
Okay. But it can be handled separately because almost every for loops have data type declarations. So it will be a big change which should be handled separately.
And this "for" loop was not newly implemented, it was moved from a different location of the same file.